### ISSUE TITLE
question: how to use conditional-if with class field

### ISSUE THREAD

**Issue by aloshai** [typestack/class-validator#1614]
I want to determine when other fields should be validated using conditional-if. If the type is folder, the value field cannot be defined. If the type is file, the value will have to be defined.

My code:
![image](https://user-images.githubusercontent.com/39468199/162635999-85dd76c1-bf7a-489c-a60b-eab345727877.png)

My TreeTypeEnum: 
![image](https://user-images.githubusercontent.com/39468199/162636154-7c79a19a-3cf5-4b37-a147-80c9d189aed6.png)

but I can still define value:
![image](https://user-images.githubusercontent.com/39468199/162636052-6ccd96bd-6de8-4b92-bd2e-1762a1164b40.png)
(whitelist, transform and forbidNonWhitelisted enabled in validate)

In short, if type is folder or file, i want they can be just define value and children field because i will save directly to database.

how can I do that?

**Comment by aloshai** [comment]
I maked to do it using the Validate and ValidateIf decarators.
